CI: Numpy Dev Build Failing

Looks like our numpy dev build also uses a pre-release (alpha) of cython.

Is this the intended behaviour?

Think this is potentially what is causing the test failures:
https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=33245&view=logs&j=3a03f79d-0b41-5610-1aa4-b4a014d0bc70&t=4d05ed0e-1ed3-5bff-dd63-1e957f2766a9 

```
attrs                     19.3.0                     py_0  
ca-certificates           2020.1.1                      0  
certifi                   2020.4.5.1               py37_0  
**cython                    3.0a1                    pypi_0    pypi**
execnet                   1.7.1                      py_0  
hypothesis                5.5.4                      py_0  
importlib_metadata        1.5.0                    py37_0  
ld_impl_linux-64          2.33.1               h53a641e_7  
libedit                   3.1.20181209         hc058e9b_0  
libffi                    3.2.1                hd88cf55_4  
libgcc-ng                 9.1.0                hdf63c60_0  
libstdcxx-ng              9.1.0                hdf63c60_0  
more-itertools            8.2.0                      py_0  
ncurses                   6.2                  he6710b0_0  
numpy                     1.19.0.dev0+8f7adad          pypi_0    pypi
openssl                   1.1.1f               h7b6447c_0  
packaging                 20.3                       py_0  
pandas                    1.1.0.dev0+1243.gd3ff12cff           dev_0    <develop>
pip                       20.0.2                   py37_1  
pluggy                    0.13.1                   py37_0  
py                        1.8.1                      py_0  
pyparsing                 2.4.6                      py_0  
pytest                    5.4.1                    py37_0  
pytest-azurepipelines     0.8.0                      py_0  
pytest-forked             1.1.3                      py_0  
pytest-xdist              1.31.0                     py_0  
python                    3.7.7           hcf32534_0_cpython  
**python-dateutil           2.8.2.dev25+gc175137          pypi_0    pypi**
pytz                      2019.3                     py_0  
readline                  8.0                  h7b6447c_0  
scipy                     1.5.0.dev0+f614064          pypi_0    pypi
setuptools                46.1.3                   py37_0  
six                       1.14.0                   py37_0  
sortedcontainers          2.1.0                    py37_0  
sqlite                    3.31.1               h7b6447c_0  
tk                        8.6.8                hbc83047_0  
wcwidth                   0.1.9                      py_0  
wheel                     0.34.2                   py37_0  
xz                        5.2.4                h14c3975_4  
zipp                      2.2.0                      py_0  
zlib                      1.2.11               h7b6447c_3
```

I think we want Cython dev to catch regressions there.

cc @scoder if you have any ideas why this would be raising.

```
>   @cython.wraparound(False)
E   TypeError: Expected dict, got bool

pandas/_libs/algos.pyx:799: TypeError
```

I'll try to recompile with Cython master to get the actual C code.

--- a/ci/deps/azure-37-numpydev.yaml
+++ b/ci/deps/azure-37-numpydev.yaml
@@ -14,8 +14,7 @@ dependencies:
   - pytz
   - pip
   - pip:
-    - cython==0.29.16
-    # GH#33507 cython 3.0a1 is causing TypeErrors 2020-04-13
+    - cython>=0.29.16
     - "git+git://github.com/dateutil/dateutil.git"
     - "-f https://7933911d6844c6c53a7d-47bd50c35cd79bd838daf386af554a83.ssl.cf2.rackcdn.com"
     - "--pre"

